### PR TITLE
Fix wrong usage of CauseAction, rename some method.

### PR DESCRIPTION
**Pull request by tianzhipeng-git**
SInce hudson.model.Run 
<pre>
 public @Nonnull List<Cause> getCauses() {
        CauseAction a = getAction(CauseAction.class);
        if (a==null)    return Collections.emptyList();
        return Collections.unmodifiableList(a.getCauses());
   }
</pre>
use just one CauseAction object to get several causes, So adding several CauseAction doesn't work.

### CODE REVIEW

File: src/main/java/com/sonyericsson/rebuild/RebuildAction.java [hunk 1]
```diff
@@ -226,12 +226,11 @@ public void parameterizedRebuild(Run currentBuild, StaplerResponse response) thr
         if (project == null) {
             return;
         }
+        System.out.println("rbx: " + (currentBuild == build));
```
Comment (GLundh) [inline src/main/java/com/sonyericsson/rebuild/RebuildAction.java]: Please avoid System.out. Use logging if text output necessary (but it is?)

Comment (tianzhipeng-git) [reply src/main/java/com/sonyericsson/rebuild/RebuildAction.java]: It's my debugging output. Been removed.

**Comment by franknarf8** [comment]
This should fix the unit tests :
```
diff --git a/src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java b/src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java
index 58d51db..730691a 100644
--- a/src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java
+++ b/src/test/java/com/sonyericsson/rebuild/RebuildValidatorTest.java
@@ -47,7 +47,6 @@ import org.jvnet.hudson.test.TestExtension;
 import org.kohsuke.stapler.DataBoundConstructor;
 
 import java.io.IOException;
-import java.util.List;
 import java.util.concurrent.ExecutionException;
 
 /**
@@ -253,25 +252,23 @@ public class RebuildValidatorTest extends HudsonTestCase {
 		while (project.isBuilding()) {
 			Thread.sleep(DELAY);
 		}
-		List<Action> actions = project.getLastCompletedBuild().getActions();
 		boolean hasRebuildCause = false;
 		boolean hasRemoteCause = false;
 		boolean hasUserIdCause = false;
-		for (Action action : actions) {
+		for (Action action : project.getLastCompletedBuild().getAllActions()) {
 			if (action instanceof CauseAction) {
-				CauseAction causeAction = (CauseAction) action;
-				if (causeAction.getCauses().get(0).getClass()
-						.equals(RebuildCause.class)) {
-					hasRebuildCause = true;
-				}
-				if (causeAction.getCauses().get(0).getClass()
-						.equals(Cause.RemoteCause.class)) {
-					hasRemoteCause = true;
-				}
-				if (causeAction.getCauses().get(0).getClass()
-						.equals(Cause.UserIdCause.class)) {
-					hasUserIdCause = true;
+				for(Cause cause : ((CauseAction)action).getCauses()) {
+					if (cause.getClass().equals(RebuildCause.class)) {
+						hasRebuildCause = true;
+					}
+					if (cause.getClass().equals(Cause.RemoteCause.class)) {
+						hasRemoteCause = true;
+					}
+					if (cause.getClass().equals(Cause.UserIdCause.class)) {
+						hasUserIdCause = true;
+					}
 				}
+
 			}
 		}
 		assertTrue("Build should have user, remote and rebuild causes",

```

**Comment by franknarf8** [comment]
@tianzhipeng-git @GLundh ,

Should I fork @tianzhipeng-git 's fork and open another pull request?

Thanks

**Pull request merged**